[ICDS Dashboard]Migration Form UCR

ShubhamBansal1997 opened the pull request:
<!--- Provide a link to the ticket or document which prompted this change -->

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This PR is made to add a new UCR for the Migration Form, this is the sub task of this JIRA ticket https://dimagi-dev.atlassian.net/browse/ICDS-1025


##### FEATURE FLAG
<!--- If this is specific to a feature flag, which one? -->

##### PRODUCT DESCRIPTION
<!--- For non-invisible changes, describe user-facing effects. -->

ShubhamBansal1997 committed:
[build]: added the Migration UCR Form

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: updated the settings.py for the migration form UCR

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor fix with name and json

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: added the abrv static-icds-cas-

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

Rohit25negi commented on settings.py:
```diff
@@ -1881,6 +1881,7 @@
     os.path.join('custom', 'icds_reports', 'ucr', 'data_sources', 'dashboard', 'daily_feeding_forms.json'),
     os.path.join('custom', 'icds_reports', 'ucr', 'data_sources', 'primary_private_school_form_ucr.json'),
     os.path.join('custom', 'icds_reports', 'ucr', 'data_sources', 'cbe_form.json'),
+    os.path.join('custom', 'icds-reports', 'ucr', 'data_sources', 'migrations_form.json'),
```
hey, I think problem is here. Its not `icds-reports`, Its `icds_reports`

Rohit25negi reviewed (changes requested):
I think problem is here.

ShubhamBansal1997 committed:
[fix]: Minor fixes for the migration ucr json
- `icds-reports` -> `icds_reports` Typo fix
- Removed the `awc_id` key in the tests
- Resovled Formatting issue in the `migration_form.json`

- Github Issue:
https://dimagi-dev.atlassian.net/browse/ICDS-1025

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

czue commented:
I don't think I have the right context to review this so will defer to @Rohit25negi

Rohit25negi commented on custom/icds_reports/ucr/data_sources/migrations_form.json:
```diff
@@ -0,0 +1,179 @@
+        "column_id": "state_id",
+        "type": "expression"
+      },
+			{
```
nitpick: can you fix the formating

Rohit25negi reviewed (changes requested):
Just formating nits! Else looks good. 
We might also need to consider the migrated_in case.

Rohit25negi reviewed (changes requested):
can you also make this UCR distributed. If you want to see the example. you can refer https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/ucr/data_sources/adolescent_girl_register_form_ucr.json#L188

ShubhamBansal1997 committed:
[fix]: make the ucr distributed as well fix indent issues

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: Made supervisior id is_primary true and is_nullable false

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

Rohit25negi commented on custom/icds_reports/ucr/data_sources/migrations_form.json:
```diff
@@ -0,0 +1,202 @@
+        "is_primary_key": true,
+        "is_nullable": false,
+        "column_id": "supervisor_id",
+        "create_index": true,
```
I think this property is not required here as its creating  a primary key on supervisor_id and timeend.

ShubhamBansal1997 committed:
[fix]: remove the property create index from the Supervisor Id

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>
